api_end_end_testing/src/test/java/tds/student/api/tests/StudentApiTests.java

```diff
@@ -0,0 +1,726 @@
+package tds.student.api.tests;
+
+import static com.jayway.restassured.RestAssured.*;
+import static org.hamcrest.Matchers.*;
+
+import com.jayway.restassured.http.ContentType;
+import org.testng.annotations.Test;
+import tds.base.BaseUri;
+import tds.student.api.model.StudentDeleteBatchInfo;
+import tds.student.api.model.StudentInfo;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/*
+ * This class tests Student API endpoints that creates, updates and deletes a student
+ */
+public class StudentApiTests extends BaseUri {
+    private String uriLocation = "/rest/external/student/";
+
+    /*
+     * Test of Create Student, HTTP POST of /api/external/student/{stateCode}, 201 success item created
+     * Test of Get Student, HTTP GET of /api/external/student/{stateCode}/{ssid}, 200 success item found
+     */
+    private StudentInfo createStudent(String ssid, String stateAbbrev) {
+        StudentInfo studentInfo = new StudentInfo(
```

Comment:
When I have these complex objects in my tests I usually have a test builder.  This [article](https://www.javacodegeeks.com/2013/06/builder-pattern-good-for-code-great-for-tests.html) gives a good example.  It is especially nice because I find in my tests there are only a couple fields I want to change or set.